probatus/missing/imputation.py

```diff
@@ -0,0 +1,318 @@
+    """
+    Comparison of various imputation stragegies that can be used for imputation 
+    of missing values. 
+    The aim of this class is to present the model performance based on imputation
+    strategies and choosen model.
+    For models like XGBoost & LighGBM which have capabilities to handle misisng values by default
+    the model performance with no imputation will be shown as well.
+    The missing values categorical features are filled with `missing` and an missing indicator is
+    added.
+
+    Usage E.g.
+    ```python
+
+    #Import the class
+    from probatus.missing.imputation import CompareImputationStrategies
+    #Create the strategies.
+    strategies = {
+       'Simple Median Imputer' : SimpleImputer(strategy='median',add_indicator=True),
+       'Simple Mean Imputer' : SimpleImputer(strategy='mean',add_indicator=True),
+       'Iterative Imputer'  : IterativeImputer(add_indicator=True,n_nearest_features=5,
+       sample_posterior=True),
+       'KNN' : KNNImputer(n_neighbors=3)
+    #Create a classifier.
+    clf = lgb.LGBMClassifier()
+    #Create the comparision of the imputation strategies.
+    cmp = CompareImputationStrategies(
+        clf=clf,
+        strategies=strategies,
+        cv=5,
+        model_na_support=True)
+
+    cmp.fit_compute(X_missing,y)
+    #Plot the results.
+    cmp.plot()
+
+    <img src="../img/imputation_comparision.png" width="500" />
+   }
+
+    ```
+
+    """
+    def __init__(self,clf,strategies,scoring='roc_auc',cv=None,model_na_support=True,n_jobs=-1,verbose=0,
+                random_state=None):
+        """
+        Initialise the class.
+
+        Args :
+            clf(model object):
+                A binary classification model, that will used to evaluate various imputation strategies.
+
+            strategies (dictionary of sklearn.impute objects):
+                Dictionary containing the sklearn.impute objects.
+                e.g.
+                
+                strategies = {'KNN' : KNNImputer(n_neighbors=3),
+                'Simple Median Imputer' : SimpleImputer(strategy='median',add_indicator=True),
+                'Iterative Imputer'  : IterativeImputer(add_indicator=True,n_nearest_features=5,
+                sample_posterior=True)}
+                This allows you to have fine grained control over the imputation method.
+
+            scoring (string, list of strings, probatus.utils.Scorer or list of probatus.utils.Scorers, optional):
+                Metrics for which the score is calculated. It can be either a name or list of names metric names and
+                needs to be aligned with predefined [classification scorers names in sklearn](https://scikit-learn.org/stable/modules/model_evaluation.html).
+                Another option is using probatus.utils.Scorer to define a custom metric.
+
+            model_na_support(boolean): default True
+                If the classifier supports missing values by default e.g. LightGBM,XGBoost etc. 
```

Comment:
I would say the default value should be False. This way the module is not going to raise exception for Logistic regression with default setting

Reply:
Done.